### PR TITLE
Improve `AccessManaged` and `AuthorityUtils` tests

### PR DESCRIPTION
**Pull request by ernestognw**
Reaches 100% coverage in `AccessManaged` and `AuthorityUtils` tests

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: test/access/manager/AccessManaged.test.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    });
+
+    describe('when role is granted with execution delay', function () {
+      beforeEach(async function () {
+        const executionDelay = web3.utils.toBN(911);
+        await this.authority.$_grantRole(this.role, roleMember, 0, executionDelay);
+      });
+
+      it('reverts if the operation is not scheduled', async function () {
+        const calldata = await this.managed.contract.methods[method]().encodeABI();
```
Comment (Amxx) [inline test/access/manager/AccessManaged.test.js:61]: This should not require an await, does it ?
```suggestion
        const calldata = this.managed.contract.methods[method]().encodeABI();
```

File: test/access/manager/AuthorityUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        const { immediate, delay } = await this.mock.$canCallWithDelay(
+          this.authority.address,
+          user,
+          other,
+          '0x12345678',
+        );
+        expect(immediate).to.equal(this.immediate);
+        expect(delay).to.be.bignumber.equal(this.delay);
+      });
+    });
```
Comment (Amxx) [inline test/access/manager/AuthorityUtils.test.js:72]: This is actually not something we expect to see. Its ok to test it, but then we should also test `false`, `delay`

Maybe we should have two for loops:
```
for (const immediate of [ true, false ]) {
  for (const delay of [ 0, 42 ].map(web3.utils.toBN)) {
    it(`returns (immediate=${immediate}, delay=${delay})`, async function () {
      const authority = await AuthorityDelayMock.new();
      await authority._setImmediate(this.immediate);
      await authority._setDelay(delay);
      const result = await this.mock.$canCallWithDelay(
          this.authority.address,
          user,
          other,
          '0x12345678',
        );
        expect(result.immediate).to.equal(immediate);
        expect(result.delay).to.be.bignumber.equal(delay);
    }
  }
}
```

**Review by Amxx — DISMISSED** [review]
LGTM

2 minor non-blocking comment.

File: test/access/manager/AccessManaged.test.js (hunk 1 above)
Comment (Amxx) [reply test/access/manager/AccessManaged.test.js:61]: there are other calls to `encodeABI()` that this may apply to.

**Review by frangio — APPROVED** [review]

**Pull request merged**